modules/builtin/src/backend/prompts/string.ts

```diff
@@ -0,0 +1,61 @@
+import { ExtractionResult, IO, Prompt, PromptConfig, ValidationResult } from 'botpress/sdk'
+
+import commonFields from './common'
+
+class PromptString implements Prompt {
+  private _maxLength: boolean
+  private _regexPattern: string
+
+  constructor({ maxLength, regexPattern }) {
+    this._maxLength = maxLength
+    this._regexPattern = regexPattern
+  }
+
+  extraction(event: IO.IncomingEvent): ExtractionResult | undefined {
+    const text = event.payload.text
+    if (text) {
+      return {
+        value: text,
+        confidence: 1
+      }
+    }
+  }
+
+  async validate(value): Promise<ValidationResult> {
+    if (value == undefined) {
+      return { valid: false, message: 'Provided value is invalid' }
+    }
+
+    if (value.length > this._maxLength) {
+      return { valid: false, message: 'Text is too long' }
+    }
+
+    if (!new RegExp(this._regexPattern).test(value)) {
+      return { valid: false, message: 'Value does not match regex pattern' }
+    }
```

Comment:
I know that we added it here, but I forgot that there was a Pattern prompt which sole purpose is to make it simple to use regex, so we can split that